src/FolderSync.cpp

```diff
@@ -1642,21 +1652,37 @@ bool CFolderSync::DeletePathToTrash(const std::wstring& path)
         pfo->SetOperationFlags(flags);
         IShellItem* psiFrom = nullptr;
         hr                  = SHCreateItemFromParsingName(path.c_str(), nullptr, IID_PPV_ARGS(&psiFrom));
+        if ((hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) || (hr == HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND)))
+        {
+            pfo->Release();
+            return true;
+        }
         if (SUCCEEDED(hr))
         {
+            // Why do "if (SUCCEEDED(hr))" twice in a row?
```

Comment:
the second `if (SUCCEEDED(hr))...` is obviously redundant. Could you amend your pull request with this?

Reply:
Done.